aiobotocore/eventstream.py

```diff
@@ -0,0 +1,54 @@
+from botocore.eventstream import EventStream, EventStreamBuffer
+
+
+class AioEventStream(EventStream):
+    def __init__(self, *args, **kwargs):
+        super(AioEventStream, self).__init__(*args, **kwargs)
+        self._buffer = EventStreamBuffer()
+
+    def _create_raw_event_generator(self):
+        return self._raw_stream.iter_chunks()
+
+    def __iter__(self):
+        raise NotImplementedError('Use async-for instead')
+
+    def __aiter__(self):
+        return self
+
+    async def __anext__(self):
+        # Basically this gets chunks of data from a stream,
+        # pumps them into a buffer class that does some error
+        # checking, splitting and then returns results
+        while True:
+            try:
+                chunk = await self._event_generator.__anext__()
```

Comment:
I think you can use the `async_generator` module I added to make this more similar to the sync version, see paginate.py for usage